operations/security_and_authentication/Change_NCN_Image_Root_Password_and_SSH_Keys_on_PIT_Node.md

```diff
@@ -1,122 +1,199 @@
 
-          SQUASHFS_ROOT_PW_HASH    If set to the encrypted hash for a root password,
-                                   this hash will be injected into /etc/shadow in the
-                                   squashfs image and there will be no interactive prompt
-                                   to set it. When setting this variable, be sure to use
-                                   single quotes (') to ensure any '$' characters are not
-                                   interpreted.
+#### Use PIT node password
 
-          DEBUG                    If set, the script will be run with 'set -x'
-   ```
+If wanting to use the same `root` user password that is being used on the PIT node where this procedure is being run, then
+the following command can be used to set the `SQUASHFS_ROOT_PW_HASH` variable.
 
-   The following example references a directory with existing keys and setting the timezone to
-   `America/Chicago`. This example will prompt the administrator to enter a root password after
-   each squashed image is unsquashed.
+```bash
+pit# export SQUASHFS_ROOT_PW_HASH=$(awk -F':' /^root:/'{print $2}' < /etc/shadow)
+```
 
-   ```bash
-   pit# cd /var/www/ephemeral/data/
-   pit# ${CSM_PATH}/ncn-image-modification.sh -p -z America/Chicago \
-                                              -d /my/pre-existing/keys \
-                                              -k $(find . -name "kubernetes-*.squashfs" | sort -V | tail -1) \
-                                              -s $(find . -name "storage-ceph-*.squashfs" | sort -V | tail -1)
-   ```
+#### Enter password and generate hash
 
-   The following example generates new keys with an empty passphrase, and the
-   `$SQUASHFS_ROOT_PW_HASH` variable set. This variable will be set to reuse the same root
-   password hash that exists on the PIT node. This example will not prompt the administrator for
-   any input after it is invoked.
+The following script can be used to manually enter a new password, and then generate its hash.
 
-   ```bash
-   pit# cd /var/www/ephemeral/data/
-   pit# export SQUASHFS_ROOT_PW_HASH=$(awk -F':' /^root:/'{print $2}' < /etc/shadow)
-   pit# ${CSM_PATH}/ncn-image-modification.sh -p \
-                                              -t rsa \
-                                              -N "" \
-                                              -k $(find . -name "kubernetes-*.squashfs" | sort -V | tail -1) \
-                                              -s $(find . -name "storage-ceph-*.squashfs" | sort -V | tail -1)
-   ```
+> The script uses `read -s` to prevent the password from being echoed to the screen or saved
+> in the shell history. It unsets the plaintext password variables at the end, so that only
+> the hash is preserved.
 
-   The script will save the original SquashFS images in `./{k8s,ceph}/old`. The new image filenames will
-   have a `secure-` prefix. The `initrd` and kernel will retain their original filenames.
+```bash
+pit# \
```

Comment:
Can this prompt (and any others) move out of the snippet?

Reply:
Not for 1.2, I don't think, but for the main backport they will.